apps/tangle-dapp/data/DelegationsPayouts/usePayouts.ts

```diff
@@ -0,0 +1,288 @@
+
+                              const nominatorTotalRewardFormatted =
+                                await formatTokenBalance(
+                                  new u128(
+                                    apiPromise.registry,
+                                    Math.floor(nominatorTotalReward)
+                                  )
+                                );
+
+                              const validatorIdentity =
+                                await getValidatorIdentity(validator);
+
+                              const validatorNominators = await Promise.all(
+                                eraStaker.others.map(async (nominator) => {
+                                  const nominatorIdentity =
+                                    await getValidatorIdentity(
+                                      nominator.who.toString()
+                                    );
+
+                                  return {
+                                    address: nominator.who.toString(),
+                                    identity: nominatorIdentity ?? '',
+                                  };
+                                })
+                              );
+
+                              if (
+                                validatorTotalStakeFormatted &&
+                                validatorTotalRewardFormatted &&
+                                nominatorTotalRewardFormatted
+                              ) {
+                                return {
+                                  era,
+                                  validator: {
+                                    address: validator,
+                                    identity: validatorIdentity ?? '',
+                                  },
+                                  validatorTotalStake:
+                                    validatorTotalStakeFormatted,
+                                  nominators: validatorNominators,
+                                  validatorTotalReward:
+                                    validatorTotalRewardFormatted,
+                                  nominatorTotalReward:
+                                    nominatorTotalRewardFormatted,
+                                  status: 'unclaimed',
+                                };
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  })
+                  .filter(
+                    (payout): payout is Promise<Payout> => payout !== undefined
+                  );
+              });
+
+              if (isMounted) {
+                const validatorPayouts = await Promise.all(
+                  validatorPayoutsPromises
+                );
+
+                setPayouts(
+                  validatorPayouts
+                    .filter((payout) => payout !== undefined)
+                    .sort((a, b) => Number(a.era) - Number(b.era))
```

Comment:
If `era` is a `BN` or its derivative, this conversion may fail for larger values of `BN`. Any other way of doing this sorting comparison?